subgraphs/dopex/src/entities/pool.ts

```diff
@@ -0,0 +1,558 @@
+    }
+    pool.inputTokenBalances = [BIGINT_ZERO];
+    pool.inputTokenWeights = [BIGDECIMAL_HUNDRED];
+    pool.createdTimestamp = event.block.timestamp;
+    pool.createdBlockNumber = event.block.number;
+
+    // Tokens
+    pool.outputToken = null;
+    pool.rewardTokens = [];
+
+    pool.fees = createPoolFees(poolAddress);
+
+    pool.createdTimestamp = event.block.timestamp;
+    pool.createdBlockNumber = event.block.number;
+
+    // Quantitative Revenue Data
+    pool.totalValueLockedUSD = BIGDECIMAL_ZERO;
+    pool.cumulativeSupplySideRevenueUSD = BIGDECIMAL_ZERO;
+    pool.cumulativeProtocolSideRevenueUSD = BIGDECIMAL_ZERO;
+    pool.cumulativeTotalRevenueUSD = BIGDECIMAL_ZERO;
+
+    pool.cumulativeEntryPremiumUSD = BIGDECIMAL_ZERO;
+    pool.cumulativeExitPremiumUSD = BIGDECIMAL_ZERO;
+    pool.cumulativeTotalPremiumUSD = BIGDECIMAL_ZERO;
+    pool.cumulativeDepositPremiumUSD = BIGDECIMAL_ZERO;
+    pool.cumulativeWithdrawPremiumUSD = BIGDECIMAL_ZERO;
+    pool.cumulativeTotalLiquidityPremiumUSD = BIGDECIMAL_ZERO;
+
+    pool.cumulativeVolumeUSD = BIGDECIMAL_ZERO;
+    pool.cumulativeExercisedVolumeUSD = BIGDECIMAL_ZERO;
+    pool.cumulativeClosedVolumeUSD = BIGDECIMAL_ZERO;
+    pool.openInterestUSD = BIGDECIMAL_ZERO;
+    pool.putsMintedCount = INT_ZERO;
+    pool.callsMintedCount = INT_ZERO;
+    pool.contractsMintedCount = INT_ZERO;
+    pool.contractsTakenCount = INT_ZERO;
+    pool.contractsExpiredCount = INT_ZERO;
+    pool.contractsExercisedCount = INT_ZERO;
+    pool.contractsClosedCount = INT_ZERO;
+    pool.openPositionCount = INT_ZERO;
+    pool.closedPositionCount = INT_ZERO;
+
+    // Quantitative Token Data
+    pool.outputTokenSupply = BIGINT_ZERO;
+    pool.outputTokenPriceUSD = BIGDECIMAL_ZERO;
+    pool.stakedOutputTokenAmount = BIGINT_ZERO;
+    pool.rewardTokenEmissionsAmount = [];
+    pool.rewardTokenEmissionsUSD = [];
+
+    pool._cumulativeOpenInterestUSD = BIGDECIMAL_ZERO;
+    pool._cumulativeOpenPositionCount = INT_ZERO;
+    pool._cumulativeDepositedVolumeUSD = BIGDECIMAL_ZERO;
+    pool._cumulativeWithdrawVolumeUSD = BIGDECIMAL_ZERO;
+    pool._cumulativeVolumeByTokenAmount = [BIGINT_ZERO];
+    pool._cumulativeVolumeByTokenUSD = [BIGDECIMAL_ZERO];
+    pool._cumulativeDepositedVolumeByTokenAmount = [BIGINT_ZERO];
+    pool._cumulativeDepositedVolumeByTokenUSD = [BIGDECIMAL_ZERO];
+    pool._cumulativeWithdrawVolumeByTokenAmount = [BIGINT_ZERO];
+    pool._cumulativeWithdrawVolumeByTokenUSD = [BIGDECIMAL_ZERO];
+    pool._currentEpoch = BIGINT_ZERO;
+    pool._lastSnapshotDayID = INT_ZERO;
+    pool._lastSnapshotHourID = INT_ZERO;
+    pool._lastUpdateTimestamp = BIGINT_ZERO;
+
+    // update number of pools
+    protocol.totalPoolCount += INT_ONE;
+    protocol.save();
+
+    pool.save();
+  }
+
+  return pool;
+}
+
+export function increasePoolVolume(
+  event: ethereum.Event,
+  pool: LiquidityPool,
+  sizeUSDDelta: BigDecimal,
+  collateralAmountDelta: BigInt,
+  collateralUSDDelta: BigDecimal,
```

Comment:
There are unique concepts for derivative trading.

For example, a user want to sell a call to be able to buy 1 BTC at the current 25K price one month later. In this scenario, the sizeUSD is 1BTC * BTCPrice = 25K. But in order to do so, he just needs to deposit right now 1 ETH as collateral, whose value is 1 ETH * ETHPrice = 1.5K (Assuming ETH is 1.5K now) 

So for derivative, there are normal two kind of values. One is the intended trading size, which is called sizeUSD in the codes, while the other is the collateral size, which is called collateralUSD in the codes.

Reply:
Oh, understood. Can you leave a comment on the function to have that for future reference?

Reply:
Comment added at the latest commit 6f93880b268e676ca1ed8cb47107040fab0dfc66

For dopex and other derivative protocols, they usually show sizeUSD as volume in order to make the volumes data bigger as the sizeUSD is much bigger (> 10) than collateralUSD, as show in the screenshot below.
<img width="761" alt="image" src="https://user-images.githubusercontent.com/28422812/226338691-1e738586-755a-496a-af1b-65bf9fa85707.png">

But in reality, the token in the pool is the collateral, not the trading token, so I use the collateralUSD to measure the token amount and volume.

Reply:
Interesting 🤔 

I think we should keep it consistent. It is fine if our methodology is different than others IMO, but the meaning 
 of "volume" should be consistent throughout the subgraph.

If what we call volume is the nominal volume, then let's use the nominal volume everywhere.

So if you are using sizeUSD for the protocol volume, use sizeUSD for the pool volume too. An alternative is to have fields for both values.

Reply:
Agreed, we should remain consistent. And the schema says "notional" on all volume fields.

So from what you said @corerouter that would mean we care about adding `sizeDeltaUSD` to each volume field, and `collateralDeltaUSD` is not necessary.

Like @jaimehgb the question would be, do we want to add a field for collateral volume? And I am going to defer that to @ishraq8 :)

Reply:
I agree, for derivatives, volume refers to notional volume. so the `sizeUSD` should be used for volume throughout options. 

I like the idea of `collateralVolume` at the pool level. if you guys agree i can add it to the schema.

Reply:
added PR: https://github.com/messari/subgraphs/pull/1967

Reply:
I will upgrade the subgraph once the PR is approved. Thanks!

Reply:
Fix the issue and upgrade the subgraph with schema updated at https://github.com/messari/subgraphs/pull/1967 at 421b2b0e385fe5bcbf359c9871169f942666ea56 Thanks!

Reply:
Apologies, I don't think the schema changes are quite final yet. There is still some more discussion to be hashed out

Reply:
I see. I will update the subgraph in case there is further changes at at https://github.com/messari/subgraphs/pull/1967

Reply:
PR https://github.com/messari/subgraphs/pull/1967 was merged without changes since last commit of dopex subgraph. So no further changes are needed for this dopex subgraph codebase. Thanks!